### PR TITLE
Show space name in leader/part distribution of show hosts

### LINKED ISSUE #909: Show space name when running "show hosts"
```
(user@127.0.0.1) [(none)]> show hosts
================================================================================================
| Ip            | Port  | Status | Leader count | Leader distribution | Partition distribution |
================================================================================================
| 192.168.8.210 | 34600 | online | 0            |                     | space 1: 3             |
------------------------------------------------------------------------------------------------
| 192.168.8.210 | 34700 | online | 2            | space 1: 2          | space 1: 3             |
------------------------------------------------------------------------------------------------
| 192.168.8.210 | 34500 | online | 1            | space 1: 1          | space 1: 3             |
------------------------------------------------------------------------------------------------
| 192.168.8.210 | 34800 | online | 0            |                     |                        |
------------------------------------------------------------------------------------------------

```

For users,  they can't understand what is 'space 1',  just show the space name directly. 
And for host with no leaders or no parts,  we'd better show zero instead of empty.

### LINKED ISSUE #925: Meta server can't work if "getLeader" blocked when show hosts
The write lock for space table will be hold during this time. 
And all requests will be hanged. 

It is a critical bug.  We should fix it ASAP.

### PR DESCRIPTION
**Pull request by critical27**
Show space name instead of space id in leader/partition distribution of `show hosts`. The screenshot of `show hosts` looks as follows:
image

close #909 
close #925

### CODE REVIEW

**Comment by critical27** [comment]
Jenkins go

**Comment by critical27** [comment]
Jenkins go

**Comment by critical27** [comment]
Jenkins go

**Comment by critical27** [comment]
Jenkins go

File: src/graph/ShowExecutor.cpp [hunk 1]
```diff
@@ -88,6 +89,13 @@ void ShowExecutor::showHosts() {
                                         "Leader distribution", "Partition distribution"};
         resp_ = std::make_unique<cpp2::ExecutionResponse>();
         resp_->set_column_names(std::move(header));
+        std::sort(hostItems.begin(), hostItems.end(), [](const auto& a, const auto& b) {
+            // sort with online/offline and ip
+            if (a.get_status() == b.get_status()) {
+                return a.hostAddr.ip < b.hostAddr.ip;
+            }
+            return a.status < b.status;
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/graph/ShowExecutor.cpp]: In this one it is the same, I will replace them.

File: src/meta/processors/partsMan/ListHostsProcessor.cpp [hunk 2]
```diff
@@ -15,17 +15,24 @@ namespace meta {
 
 void ListHostsProcessor::process(const cpp2::ListHostsReq& req) {
     UNUSED(req);
-    folly::SharedMutex::ReadHolder rHolder(LockUtils::spaceLock());
-    auto ret = allHostsWithStatus();
-    if (!ret.ok()) {
-        onFinished();
-        return;
+    std::unordered_map<GraphSpaceID, std::string> spaceIdName;
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/meta/processors/partsMan/ListHostsProcessor.cpp]: Good suggestion

File: src/meta/processors/partsMan/ListHostsProcessor.h [hunk 3]
```diff
@@ -28,9 +28,16 @@ class ListHostsProcessor : public BaseProcessor<cpp2::ListHostsResp> {
             , adminClient_(adminClient) {}
 
     /**
-     * Get all hosts with online/offline status.
+     * Get all hosts with online/offline status and partition distribution.
      * */
-    StatusOr<std::vector<cpp2::HostItem>> allHostsWithStatus();
+    StatusOr<std::vector<cpp2::HostItem>>
+    allHostsWithStatus(std::unordered_map<GraphSpaceID, std::string>& spaceIdName);
```
Comment (yixinglu) [inline src/meta/processors/partsMan/ListHostsProcessor.h]: `spaceIdName` -> `spaceIdNameMap`

File: src/meta/processors/partsMan/ListHostsProcessor.h [hunk 4]
```diff
@@ -28,9 +28,16 @@ class ListHostsProcessor : public BaseProcessor<cpp2::ListHostsResp> {
...
+
+    /**
+     * Get all hosts with storage leader distribution.
+     * */
+    void getLeaderDist(std::vector<cpp2::HostItem>& hostItems,
+                       std::unordered_map<GraphSpaceID, std::string>& spaceIdName);
```
Comment (yixinglu) [inline src/meta/processors/partsMan/ListHostsProcessor.h]: ditto

**Comment by critical27** [comment]
Jenkins go

**Comment by whitewum** [comment]
Great! Can you help to update .md file in docs/manual-EN/statement-syntax/data-definition-statements/show-syntax.md

File: src/meta/processors/partsMan/ListHostsProcessor.cpp [hunk 5]
```diff
@@ -63,13 +70,21 @@ StatusOr<std::vector<cpp2::HostItem>> ListHostsProcessor::allHostsWithStatus() {
...
         iter->next();
     }
 
     std::unordered_map<HostAddr,
-                       std::unordered_map<GraphSpaceID, std::vector<PartitionID>>> allParts;
+                       std::unordered_map<std::string, std::vector<PartitionID>>> allParts;
     for (const auto& spaceId : spaces) {
+        // get space name by space id
+        auto it = spaceIdNameMap.find(spaceId);
+        if (it == spaceIdNameMap.end()) {
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/meta/processors/partsMan/ListHostsProcessor.cpp]: Good catch. Remove redundant check. Cuz we need to scan all the partition, and the previous loop is to get all spaces, so maybe we can't combine them.

**Comment by critical27** [comment]
> docs/manual-EN/statement-syntax/data-definition-statements/show-syntax.md

updated.

File: src/storage/QueryEdgePropsProcessor.cpp [hunk 6]
```diff
@@ -62,7 +62,7 @@ void QueryEdgePropsProcessor::process(const cpp2::EdgePropRequest& req) {
     RowSetWriter rsWriter;
     std::for_each(req.get_parts().begin(), req.get_parts().end(), [&](auto& partE) {
         auto partId = partE.first;
-        kvstore::ResultCode ret;
+        kvstore::ResultCode ret = kvstore::ResultCode::SUCCEEDED;
```
Comment (darionyaphet) [inline src/storage/QueryEdgePropsProcessor.cpp]: why use `SUCCEEDED` to set ret ? Is it really necessary ?

Comment (critical27) [reply src/storage/QueryEdgePropsProcessor.cpp]: It's to suppress compilers warning of uninitialized variables

**Comment by dangleptr** [comment]
LGTM.  Please fix the conflicts

**Pull request merged**